tests/settings_tests/tests.py

```diff
@@ -511,3 +511,45 @@ def test_session_verification_deprecation_neither(self):
             warnings.filterwarnings('always')
             Settings('fake_settings_module')
         self.assertEqual(len(warn), 0)
+
+
+class TestAllowedHostsSettings(unittest.TestCase):
+    """
+    Make sure Allowed_Hosts setting raises an ImproperlyConfigured during
```

Comment:
ImproperlyConfigured should be raised if DEBUG=False and ALLOWED_HOSTS is empty.